aten/src/ATen/ATen.h

```diff
@@ -1,7 +1,9 @@
 #pragma once
 
-#if !defined(_MSC_VER) && __cplusplus < 201402L
```

Comment:
Also @ezyang mentioned some downstream libraries may still be C++14 so the headers should be C++14 compatible when possible.